test/workload_discovery/unix/util/scripts

```diff
@@ -0,0 +1,236 @@
+
+    for arg in "$@"; do
+        if echo "$arg" | grep -q "="; then
+            key=$(echo "$arg" | cut -d'=' -f1)
+            value=$(echo "$arg" | cut -d'=' -f2-)
+            echo "$key: $value" | sudo tee -a /tmp/jvm-compile/META-INF/MANIFEST.MF > /dev/null
+        fi
+    done
+    
+    cd /tmp/jvm-compile
+    jar cfm "/tmp/$JAR_NAME" META-INF/MANIFEST.MF *.class
+    sudo rm -rf /tmp/jvm-compile
+}
+
+setup_tomcat() {
+    VERSION="${1:-apache-tomcat-9.0.110}"
+    BUCKET="$2"
+    aws s3 cp "s3://$BUCKET/tomcat/$VERSION.tar.gz" "/tmp/$VERSION.tar.gz"
+    sudo tar -xzf "/tmp/$VERSION.tar.gz" -C /tmp
+    TOMCAT_DIR="/tmp/$VERSION"
+    sudo chmod -R 755 "$TOMCAT_DIR"
+}
+
+setup_kafka() {
+    VERSION="${1:-kafka_2.13-3.5.0}"
+    BUCKET="$2"
+    aws s3 cp "s3://$BUCKET/kafka/$VERSION.tgz" "/tmp/$VERSION.tgz"
+    sudo tar -xzf "/tmp/$VERSION.tgz" -C /tmp
+    KAFKA_DIR="/tmp/$VERSION"
+    sudo chmod -R 755 "$KAFKA_DIR"
+}
+
+spin_up_jvm() {
+    JAR_PATH="${1:-/tmp/test.jar}"
+    PORT="$2"
+    
+    if [ -n "$PORT" ]; then
+        java -Dcom.sun.management.jmxremote \
+             -Dcom.sun.management.jmxremote.port="$PORT" \
+             -Dcom.sun.management.jmxremote.local.only=false \
+             -Dcom.sun.management.jmxremote.authenticate=false \
+             -Dcom.sun.management.jmxremote.ssl=false \
+             -Dcom.sun.management.jmxremote.rmi.port="$PORT" \
+             -Dcom.sun.management.jmxremote.host=localhost \
+             -Djava.rmi.server.hostname=localhost \
+             -jar "$JAR_PATH" >/dev/null 2>&1 &
+    else
+        java -jar "$JAR_PATH" >/dev/null 2>&1 &
+    fi
+    echo $!
+}
+
+tear_down_jvm() {
+    PID="$1"
+    kill "$PID" 2>/dev/null
+    while kill -0 "$PID" 2>/dev/null; do
+        sleep 0.1
+    done
+}
+
+spin_up_tomcat() {
+    TOMCAT_DIR="$1"
+    PORT="$2"
+    
+    export CATALINA_BASE="$TOMCAT_DIR"
+    if [ -n "$PORT" ]; then
+        export CATALINA_OPTS="-Dcom.sun.management.jmxremote.port=$PORT -Dcom.sun.management.jmxremote.authenticate=false -Dcom.sun.management.jmxremote.ssl=false"
+    fi
+    
+    sudo -E "$TOMCAT_DIR/bin/startup.sh"
+    sleep 10
+}
+
+tear_down_tomcat() {
+    TOMCAT_DIR="$1"
+    unset CATALINA_OPTS CATALINA_BASE
+    sudo "$TOMCAT_DIR/bin/shutdown.sh"
+}
+
+spin_up_kafka() {
+    KAFKA_DIR="$1"
+    VERSION="$2"
+    PORT="$3"
+    
+    MAJOR_VERSION=$(echo "$VERSION" | cut -d'-' -f2 | cut -d'.' -f1)
+    if [ "$MAJOR_VERSION" -lt 4 ]; then
+        sudo -E "$KAFKA_DIR/bin/zookeeper-server-start.sh" "$KAFKA_DIR/config/zookeeper.properties" &
+        sleep 10
+        CONFIG_FILE="$KAFKA_DIR/config/server.properties"
+    else
+        CONFIG_FILE="$KAFKA_DIR/config/controller.properties"
+        UUID=$("$KAFKA_DIR/bin/kafka-storage.sh" random-uuid)
+        "$KAFKA_DIR/bin/kafka-storage.sh" format -t "$UUID" -c "$CONFIG_FILE" --standalone
+    fi
+    
+    if [ -n "$PORT" ]; then
```

Comment:
Have we considered validating the `KAFKA_JMX_PORT` environment variable as well?